crates/api_crud/src/user/create.rs

```diff
@@ -509,35 +523,44 @@ async fn oauth_request_access_token(
   context: &Data<LemmyContext>,
   oauth_provider: &OAuthProvider,
   code: &str,
+  pkce_code_verifier: Option<&str>,
   redirect_uri: &str,
 ) -> LemmyResult<TokenResponse> {
+  let form = [
+    ("client_id", &*oauth_provider.client_id),
+    ("client_secret", &*oauth_provider.client_secret),
+    ("code", code),
+    ("grant_type", "authorization_code"),
+    ("redirect_uri", redirect_uri),
+  ];
+
+  let form = match pkce_code_verifier.as_deref() {
+    Some(code_verifier) => [&form[..], &[("code_verifier", code_verifier)]].concat(),
+    None => form.to_vec(),
+  };
+
   // Request an Access Token from the OAUTH provider
   let response = context
     .client()
     .post(oauth_provider.token_endpoint.as_str())
     .header("Accept", "application/json")
-    .form(&[
-      ("grant_type", "authorization_code"),
-      ("code", code),
-      ("redirect_uri", redirect_uri),
-      ("client_id", &oauth_provider.client_id),
-      ("client_secret", &oauth_provider.client_secret),
-    ])
+    .form(&form[..])
     .send()
     .await;
 
   let response = response.map_err(|_| LemmyErrorType::OauthLoginFailed)?;
+
   if !response.status().is_success() {
-    Err(LemmyErrorType::OauthLoginFailed)?;
+    dbg!(response.bytes().await);
```

Comment:
It'd probably be better to include this in the error message if its vital. Otherwise just remove this line. Ideally after you've tested this and it works, you can remove it.